cdap-common/src/main/java/co/cask/cdap/common/app/MainClassLoader.java

```diff
@@ -49,8 +52,10 @@
 
   private static final String DATASET_CLASS_NAME = Dataset.class.getName();
   private final DatasetClassRewriter datasetRewriter;
+  private final AuthEnforceRewriter authEnforceRewriter;
   private final Function<String, URL> resourceLookup;
   private final Map<String, Boolean> cache;
+  private RewritesNeeded rewritesNeeded;
```

Comment:
Using this field is a bit unsafe, since classloading can be concurrent if concurrent classloading attribute is enabled.